### PR TITLE
Get event_data working on shiny modules

### PR DESCRIPTION
**Pull request by cpsievert**
This PR addresses #659. 

I see at least two ways to approach this problem:

(1) Have `plot_ly()` and `event_data()` both accept/query a shiny session object and enforce namespacing under-the-hood using that object. This will make it impossible to obtain data from events originating in another module. This seems to be the main motivation behind using modules anyway, so I think this is the preferred approach, and you can see the implementation in https://github.com/ropensci/plotly/pull/700/commits/d2257d4747f4a5361e55c70c971579fd7b83b24b

(2) Have `event_data()` always look in the global scope and leave it to the user to "namespace" the `source` arguments. This would conflict with the philosophy of using modules, but it wouldn't be that onerous on users, and saves us some hacks.

See [here](https://github.com/ropensci/plotly/blob/fix/ns/inst/examples/plotlyShinyModules/app.R) for a toy example. Any thoughts @jcheng5? @happyshows?

### CODE REVIEW

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/156589392 

 Detected 4 differences -> 

 http://cpsievert.github.io/plotly-test-table/c2a47bd/jitter-basic
http://cpsievert.github.io/plotly-test-table/c2a47bd/plotly-alpha-blending
http://cpsievert.github.io/plotly-test-table/c2a47bd/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/c2a47bd/scatterplots-jitter

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/156666753 

 Detected 4 differences -> 

 http://cpsievert.github.io/plotly-test-table/2f96afe/jitter-basic
http://cpsievert.github.io/plotly-test-table/2f96afe/plotly-alpha-blending
http://cpsievert.github.io/plotly-test-table/2f96afe/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/2f96afe/scatterplots-jitter

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/156668966 

 Detected 4 differences -> 

 http://cpsievert.github.io/plotly-test-table/18b1a03/jitter-basic
http://cpsievert.github.io/plotly-test-table/18b1a03/plotly-alpha-blending
http://cpsievert.github.io/plotly-test-table/18b1a03/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/18b1a03/scatterplots-jitter

File: inst/examples/plotlyShinyModules/app.R [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ui <- fluidPage(
+  reusableUI("one"),
+  reusableUI("two")
+)
+
+server <- function(input, output, session) {
+  callModule(viz, "one", session = session)
+  callModule(viz, "two", session = session)
+}
+
```
Comment (cpsievert) [inline inst/examples/plotlyShinyModules/app.R]: @jcheng5 this example seems to still work without passing the session to `plot_ly()`/`event_data()`. Would u still suggest doing so?

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/156668966 

 Detected 4 differences -> 

 http://cpsievert.github.io/plotly-test-table/18b1a03/jitter-basic
http://cpsievert.github.io/plotly-test-table/18b1a03/plotly-alpha-blending
http://cpsievert.github.io/plotly-test-table/18b1a03/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/18b1a03/scatterplots-jitter

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/156868461 

 Detected 4 differences -> 

 http://cpsievert.github.io/plotly-test-table/88a5dad/jitter-basic
http://cpsievert.github.io/plotly-test-table/88a5dad/plotly-alpha-blending
http://cpsievert.github.io/plotly-test-table/88a5dad/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/88a5dad/scatterplots-jitter

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/157732283 

 Detected 4 differences -> 

 http://cpsievert.github.io/plotly-test-table/11cb061/jitter-basic
http://cpsievert.github.io/plotly-test-table/11cb061/plotly-alpha-blending
http://cpsievert.github.io/plotly-test-table/11cb061/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/11cb061/scatterplots-jitter

File: R/plotly.R [hunk 2]
```diff
@@ -119,6 +120,10 @@ plot_ly <- function(data = data.frame(), ..., type = NULL,
   id <- new_id()
   # avoid weird naming clashes
   plotlyVisDat <- data
+  # automatically namespace source
+  if (!is.null(session)) {
+    source <- session$ns(source)
```
Comment (jcheng5) [inline R/plotly.R]: Why do you want the source namespaced? Don't you want to be able to have multiple modules share the same source?

File: inst/examples/plotlyShinyModules/app.R (hunk 1 above)
Comment (jcheng5) [inline inst/examples/plotlyShinyModules/app.R]: No, you generally don't want to be explicitly passing the session--just let it use the default.

Comment (jcheng5) [inline inst/examples/plotlyShinyModules/app.R]: Same goes for all the plot_ly and event_data calls, they shouldn't need explicit session arguments passed to them either.

Comment (jcheng5) [inline inst/examples/plotlyShinyModules/app.R]: Oh I think what I said to you last week was that you should explicitly pass the *source*, not the session. So `viz` would take a `source` argument that's passed to `plot_ly` and `event_data`; that way, the main UI can decide whether `reusableUI("one")` and `reusableUI("two")` are linked together or separately.

File: R/plotly.R (hunk 2 above)
Comment (cpsievert) [inline R/plotly.R]: I was thinking namespacing would be imposed by default by plotly, but I think I'm now in favor of the approach you describe below

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/157926437 

 Detected 4 differences -> 

 http://cpsievert.github.io/plotly-test-table/11cb061/jitter-basic
http://cpsievert.github.io/plotly-test-table/11cb061/plotly-alpha-blending
http://cpsievert.github.io/plotly-test-table/11cb061/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/11cb061/scatterplots-jitter

File: inst/htmlwidgets/plotly.js [hunk 3]
```diff
@@ -62,10 +62,8 @@ HTMLWidgets.widget({
           attachKey("key");
           return obj; 
         });
-        Shiny.onInputChange(
-          ".clientValue-" + eventType + "-" + x.source, 
-          JSON.stringify(d)
-        );
+        var src = ".clientValue-" + eventType + "-" + x.source;
+        Shiny.onInputChange(src, JSON.stringify(d));
```
Comment (jcheng5) [inline inst/htmlwidgets/plotly.js]: Is there a reason you're doing `JSON.stringify()` and `jsonlite::fromJSON` explicitly? If you just pass `d` from here, and don't JSON-decode from R, I think it should work fine--we do the JSON encoding for you.

Comment (jcheng5) [inline inst/htmlwidgets/plotly.js]: That's fair. It does mean you are sending double-encoded JSON data across the wire which will bloat things up a little, but I guess these are pretty small packets anyway.

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/157935301 

 Detected 4 differences -> 

 http://cpsievert.github.io/plotly-test-table/6d598a9/jitter-basic
http://cpsievert.github.io/plotly-test-table/6d598a9/plotly-alpha-blending
http://cpsievert.github.io/plotly-test-table/6d598a9/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/6d598a9/scatterplots-jitter

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/157935301 

 Detected 4 differences -> 

 http://cpsievert.github.io/plotly-test-table/6d598a9/jitter-basic
http://cpsievert.github.io/plotly-test-table/6d598a9/plotly-alpha-blending
http://cpsievert.github.io/plotly-test-table/6d598a9/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/6d598a9/scatterplots-jitter

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/159177861 

 Detected 4 differences -> 

 http://cpsievert.github.io/plotly-test-table/ed5ef0e/jitter-basic
http://cpsievert.github.io/plotly-test-table/ed5ef0e/plotly-alpha-blending
http://cpsievert.github.io/plotly-test-table/ed5ef0e/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/ed5ef0e/scatterplots-jitter

**Comment by cpsievert** [comment]
> The message below was automatically generated after build https://travis-ci.org/ropensci/plotly/builds/159181333 

 Detected 4 differences -> 

 http://cpsievert.github.io/plotly-test-table/d2b531d/jitter-basic
http://cpsievert.github.io/plotly-test-table/d2b531d/plotly-alpha-blending
http://cpsievert.github.io/plotly-test-table/d2b531d/scatterplots-geom_jitter
http://cpsievert.github.io/plotly-test-table/d2b531d/scatterplots-jitter

**Pull request merged**